system/include/emscripten/threading.h

```diff
@@ -299,6 +299,54 @@ int emscripten_dispatch_to_thread_async_(pthread_t target_thread,
   emscripten_dispatch_to_thread_async_(                                        \
     (target_thread), (sig), (void*)(func), (satellite), ##__VA_ARGS__)
 
+// Similar to emscripten_dispatch_to_thread, but waits on the dispatching thread
+// until `func` has returned on the target thread.
+int emscripten_dispatch_to_thread_sync_ptr(pthread_t target_thread,
+                                           void (*func)(void*),
+                                           void* arg);
+int emscripten_dispatch_to_thread_sync_args(pthread_t target_thread,
```

Comment:
The other function we have in this file that do sync and async versions are of the following form:

```
emscripten_sync_run_in_main_runtime_thread
emscripten_async_run_in_main_runtime_thread
```

Should these therefore be called `emscripten_sync_dispatch_to_thread_*`?

I also wonder if they should be called something completely different such as `emscripten_run_in_thread` (really should `emscripten_run_on_thread` I guess).    The word dispatch sounds inherently async to me.. maybe maybe I'm wrong about that?